xfields/beam_elements/beambeam3d.py

```diff
@@ -257,6 +312,9 @@ def __init__(self,
         self.num_slices_other_beam = n_slices
         self.slices_other_beam_num_particles = self._arr2ctx(np.array(
                                     slices_other_beam_num_particles))
+        self.slices_other_beam_num_macroparticles = self._arr2ctx(np.array(
+                                    slices_other_beam_num_macroparticles))
+
```

Comment:
If the slice contains 0 particles, the mean and sigma are 0. If the slice contains 1 particle, the sigma is 0. If the sigma is 0 then the cerrf returns nans, so the field will be nan. Instead of the if on the macroparticles I could add an if on sigma > 0, but there has to be an if anyways.

Reply:
see [4 hours ago](https://github.com/xsuite/xfields/pull/50#discussion_r1003030751)